### ISSUE TITLE
[JENKINS-56450] MSBuild parser fails when compiling with /MP

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/warnings-ng-plugin#2795]
<p>When compiling with /MP, Visual Studio extends warnings with a note</p>

<p>    (compiling source file F:\ull\path.cpp)</p>

<p>which is parsed as part of the directory, so warnings are shown with a directory name that looks like</p>

<p>   ] (compiling source file F:\ull\path.cpp)F:\ull\path\to\directory</p>

<p>This also breaks collapsing identical warnings in headers.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/simonrichter.jpg" title="simonrichter's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=simonrichter">simonrichter</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450" target="_blank">MSBuild parser fails when compiling with /MP</a></i></summary>
<i><ul>
<li><b>assignee</b>: <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=laebi">laebi</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: warnings-ng-plugin
<li><b>label(s)</b>: help-wanted, newbie-friendly
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2023-03-15T12:21:59+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-09
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>When compiling with /MP, Visual Studio extends warnings with a note</p>

<p>    (compiling source file F:\ull\path.cpp)</p>

<p>which is parsed as part of the directory, so warnings are shown with a directory name that looks like</p>

<p>   ] (compiling source file F:\ull\path.cpp)F:\ull\path\to\directory</p>

<p>This also breaks collapsing identical warnings in headers.</p></pre>
</details>
</details>
<details><summary><i>1 attachment</i></summary>

- [dir.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/46309/dir.png)
  > ![dir.png](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-misc/refs/heads/main/attachments/46309/dir.png)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=362415&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362415">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Is there a reason to use <tt>/MP</tt>? </p>

<p>Can you please add a text based sample of some of those messages? </p>
</pre>
</details>

</ul>
</details>

<p>Is there a reason to use <tt>/MP</tt>? </p>

<p>Can you please add a text based sample of some of those messages? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/simonrichter.jpg" title="simonrichter's avatar" /> simonrichter:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=362418&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362418">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Typical lines look like</p>
<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\include\footprint_info.h(320): warning C4251: 'FOOTPRINT_ASYNC_LOADER::m_last_table': class 'std::basic_string&lt;char,std::char_traits&lt;char&gt;,std::allocator&lt;char&gt;&gt;' needs to have dll-interface to be used by clients of class 'FOOTPRINT_ASYNC_LOADER' (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\cvpcb\cvpcb_mainframe.cpp) [C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\build\cvpcb\cvpcb_kiface.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\include\xstring(4373): note: see declaration of 'std::basic_string&lt;char,std::char_traits&lt;char&gt;,std::allocator&lt;char&gt;&gt;' (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\cvpcb\cvpcb_mainframe.cpp)</pre>
</div></div>
<p>The project uses <tt>/MP</tt> because we have rather few output binaries that depend on each other, and comparatively many source files. With MSBuild's <tt>/M</tt> option, we get a maximum concurrency of 1–3, while <tt>/MP</tt> uses all 32 CPUs on our build server for about 90% of the time, reducing build time from 40 to 5 minutes.</p>
</pre>
</details>

</ul>
</details>

<p>Typical lines look like</p>


```
C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\include\footprint_info.h(320): warning C4251: 'FOOTPRINT_ASYNC_LOADER::m_last_table': class 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' needs to have dll-interface to be used by clients of class 'FOOTPRINT_ASYNC_LOADER' (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\cvpcb\cvpcb_mainframe.cpp) [C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\build\cvpcb\cvpcb_kiface.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\VC\Tools\MSVC\14.16.27023\include\xstring(4373): note: see declaration of 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\cvpcb\cvpcb_mainframe.cpp)
```
<p>The project uses <tt>/MP</tt> because we have rather few output binaries that depend on each other, and comparatively many source files. With MSBuild's <tt>/M</tt> option, we get a maximum concurrency of 1–3, while <tt>/MP</tt> uses all 32 CPUs on our build server for about 90% of the time, reducing build time from 40 to 5 minutes.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=362422&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362422">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Can you please wrap in {noformat} before and after the log?</p>
</pre>
</details>

</ul>
</details>

<p>Can you please wrap in {noformat} before and after the log?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/simonrichter.jpg" title="simonrichter's avatar" /> simonrichter:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=362423&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362423">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Done.</p>
</pre>
</details>

</ul>
</details>

<p>Done.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=362425&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362425">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I don't want to be pedantic but a parsers reads the log <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> Are the whitespace at the beginning of the 2nd line correct? (I can't change the comment on my own).</p>
</pre>
</details>

</ul>
</details>

<p>I don't want to be pedantic but a parsers reads the log <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/wink.png" height="16" width="16" align="absmiddle" alt="" border="0"/> Are the whitespace at the beginning of the 2nd line correct? (I can't change the comment on my own).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/simonrichter.jpg" title="simonrichter's avatar" /> simonrichter:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=362427&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362427">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes, these are correct. For C++ templates, this is quite common, these are in principle an extension of the previous message. In theory they can be interleaved with messages from other source files (with cl.exe /MP) and/or projects (with MSBuild /M), the latter is equivalent to the <tt><span class="error">&#91;0-9&#93;</span>+&gt;</tt> prefixes generated by VS 2015 and earlier (and some versions of 2017 IIRC). I have never seen the interleaving in effect though, only read about it in bug reports, as this seems to be timing dependent.</p>
</pre>
</details>

</ul>
</details>

<p>Yes, these are correct. For C++ templates, this is quite common, these are in principle an extension of the previous message. In theory they can be interleaved with messages from other source files (with cl.exe /MP) and/or projects (with MSBuild /M), the latter is equivalent to the <tt><span class="error">&#91;0-9&#93;</span>+></tt> prefixes generated by VS 2015 and earlier (and some versions of 2017 IIRC). I have never seen the interleaving in effect though, only read about it in bug reports, as this seems to be timing dependent.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=362428&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362428">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Ok. So one warning is expected (the note at line 2 is ignored because it starts not at position 1). </p>
</pre>
</details>

</ul>
</details>

<p>Ok. So one warning is expected (the note at line 2 is ignored because it starts not at position 1). </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/simonrichter.jpg" title="simonrichter's avatar" /> simonrichter:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=362430&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-362430">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yes. In an ideal world, the note would be accessible from the detail view as well (often, these refer to other files in the source tree), but that is probably a larger change (g++ would benefit from that as well though).</p>

<p>The minimal change required would be to ignore the "compiling source file ..." part in parentheses before the project name (which seems to be somewhat optional, at least I don't see it on every line).</p>

<p>Long logs for testing if needed:</p>
<ul>
	<li><a href="https://jenkins.simonrichter.eu/job/windows-kicad-msvc-tom/lastSuccessfulBuild/build=release,cpu=amd64,label=msvc/consoleText" class="external-link" target="_blank" rel="nofollow noopener">MSBuild serial build, with /MP</a></li>
	<li><a href="https://jenkins.simonrichter.eu/job/windows-kicad-msvc-analyze-head/lastSuccessfulBuild/build=release,cpu=amd64,label=msvc/45/consoleText" class="external-link" target="_blank" rel="nofollow noopener">MSBuild /M build, without /MP, 55,000 generated warnings</a></li>
	<li><a href="https://jenkins.simonrichter.eu/job/windows-kicad-msvc-head/lastSuccessfulBuild/build=release,cpu=amd64,label=msvc/consoleText" class="external-link" target="_blank" rel="nofollow noopener">MSBuild /M build, without /MP</a></li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Yes. In an ideal world, the note would be accessible from the detail view as well (often, these refer to other files in the source tree), but that is probably a larger change (g++ would benefit from that as well though).</p>

<p>The minimal change required would be to ignore the "compiling source file ..." part in parentheses before the project name (which seems to be somewhat optional, at least I don't see it on every line).</p>

<p>Long logs for testing if needed:</p>
<ul>
	<li><a href="https://jenkins.simonrichter.eu/job/windows-kicad-msvc-tom/lastSuccessfulBuild/build=release,cpu=amd64,label=msvc/consoleText" class="external-link" target="_blank" rel="nofollow noopener">MSBuild serial build, with /MP</a></li>
	<li><a href="https://jenkins.simonrichter.eu/job/windows-kicad-msvc-analyze-head/lastSuccessfulBuild/build=release,cpu=amd64,label=msvc/45/consoleText" class="external-link" target="_blank" rel="nofollow noopener">MSBuild /M build, without /MP, 55,000 generated warnings</a></li>
	<li><a href="https://jenkins.simonrichter.eu/job/windows-kicad-msvc-head/lastSuccessfulBuild/build=release,cpu=amd64,label=msvc/consoleText" class="external-link" target="_blank" rel="nofollow noopener">MSBuild /M build, without /MP</a></li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>laebi:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=363861&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-363861">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>So the warning message should still be:</p>

<p> <tt>'FOOTPRINT_ASYNC_LOADER::m_last_table': class 'std::basic_string&lt;char,std::char_traits&lt;char&gt;,std::allocator&lt;char&gt;&gt;' needs to have dll-interface to be used by clients of class 'FOOTPRINT_ASYNC_LOADER' (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\cvpcb\cvpcb_mainframe.cpp)</tt></p>

<p>But the directory should not contain the file path in the brackets. At the moment I do not see where the directory path is generated... But I will find out or someone gives me a hint.</p>
</pre>
</details>

</ul>
</details>

<p>So the warning message should still be:</p>

<p> <tt>'FOOTPRINT_ASYNC_LOADER::m_last_table': class 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' needs to have dll-interface to be used by clients of class 'FOOTPRINT_ASYNC_LOADER' (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\cvpcb\cvpcb_mainframe.cpp)</tt></p>

<p>But the directory should not contain the file path in the brackets. At the moment I do not see where the directory path is generated... But I will find out or someone gives me a hint.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/simonrichter.jpg" title="simonrichter's avatar" /> simonrichter:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=363864&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-363864">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>It's totally fine to remove that directory tag from the warning message.</p>

<p>The bit about the "note:" lines is a bit confusing, that is a separate issue which definitely isn't beginner level anymore, and which I've filed as <a href="https://issues.jenkins.io/browse/JENKINS-56815" title="Make &quot;note&quot; lines available in warning details" class="issue-link" data-issue-key="JENKINS-56815"><del>JENKINS-56815</del></a> now.</p>

<p>Our main problem at the moment is that the build with /MP shows close to 15,000 warnings per configuration, while the non-parallel build only has 1,500 per configuration, as many of these warnings are generated in header files, so these would be collapsed as duplicates, but the source name added to the message makes them unique.</p>
</pre>
</details>

</ul>
</details>

<p>It's totally fine to remove that directory tag from the warning message.</p>

<p>The bit about the "note:" lines is a bit confusing, that is a separate issue which definitely isn't beginner level anymore, and which I've filed as <a href="https://issue-redirect.jenkins.io/issue/56815" title="Make "note" lines available in warning details" class="issue-link" data-issue-key="<a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/56815">JENKINS-56815</a>"><del>JENKINS-56815</del></a> now.</p>

<p>Our main problem at the moment is that the build with /MP shows close to 15,000 warnings per configuration, while the non-parallel build only has 1,500 per configuration, as many of these warnings are generated in header files, so these would be collapsed as duplicates, but the source name added to the message makes them unique.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>laebi:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=364765&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-364765">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I found out that your description is not the root of the problem, but log entries like the following leads to a 'wrong' filename detection. I assume that there is a missing protection of the log-generation when several processors generate log messages in parallel:</p>

<p><tt>C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\include\footprint_info.h(320): warning C4251: 'FOOTPRINT_ASYNC_LOADER::m_last_table': class 'std::basic_string&lt;char,std::char_traits&lt;char&gt;,std::allocator&lt;char&gt;&gt;' needs to have dll-interface to be used by clients of class 'FOOTPRINT_ASYNC_LOADER' (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\pcbnew\footprint_libraries_utils.cpp)C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\include\geometry/seg.h(263): warning C4244: 'initializing': conversion from 'double' to 'SEG::ecoord', possible loss of data (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\pcbnew\hotkeys_footprint_editor.cpp) <span class="error">&#91;C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\build\pcbnew\pcbnew_kiface_objects.vcxproj&#93;</span></tt></p>

<p>I will create a PR to detect such log messages as warning on the first file / message (not able to split the line and create 2 issues).</p>
</pre>
</details>

</ul>
</details>

<p>I found out that your description is not the root of the problem, but log entries like the following leads to a 'wrong' filename detection. I assume that there is a missing protection of the log-generation when several processors generate log messages in parallel:</p>

<p><tt>C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\include\footprint_info.h(320): warning C4251: 'FOOTPRINT_ASYNC_LOADER::m_last_table': class 'std::basic_string<char,std::char_traits<char>,std::allocator<char>>' needs to have dll-interface to be used by clients of class 'FOOTPRINT_ASYNC_LOADER' (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\pcbnew\footprint_libraries_utils.cpp)C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\include\geometry/seg.h(263): warning C4244: 'initializing': conversion from 'double' to 'SEG::ecoord', possible loss of data (compiling source file C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\src\pcbnew\hotkeys_footprint_editor.cpp) <span class="error">&#91;C:\Jenkins\workspace\windows-kicad-msvc-tom\build\release\cpu\amd64\label\msvc\build\pcbnew\pcbnew_kiface_objects.vcxproj&#93;</span></tt></p>

<p>I will create a PR to detect such log messages as warning on the first file / message (not able to split the line and create 2 issues).</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>marne:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=436004&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-436004">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We are seeing this issue while using</p>
<ul>
	<li>Analysis Model API in Version 10.23.1</li>
	<li>Warnings Next Generation in Version 9.23.1</li>
</ul>


<p>That is the same issue in the same file is reported multiple times. Each time with another file in the "compiling source file" annotation.</p>

<p>Example log outputs:</p>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>133&gt;C:\jenkins\workspace\develop\file.h(340,78): warning C4239: nonstandard extension used: 'default argument': conversion from 'CSize' to 'CSize &amp;' (compiling source file main.cpp) [C:\jenkins\workspace\develop\project.vcxproj]
</pre>
</div></div>

<div class="preformatted panel" style="border-width: 1px;"><div class="preformattedContent panelContent">
<pre>175&gt;c:\jenkins\workspace\develop\file.h(76): warning C4100: "Id": Unreferenzierter formaler Parameter (Quelldatei wird kompiliert main.cpp) [C:\jenkins\workspace\develop\project.vcxproj]
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p>We are seeing this issue while using</p>
<ul>
	<li>Analysis Model API in Version 10.23.1</li>
	<li>Warnings Next Generation in Version 9.23.1</li>
</ul>


<p>That is the same issue in the same file is reported multiple times. Each time with another file in the "compiling source file" annotation.</p>

<p>Example log outputs:</p>



```
133>C:\jenkins\workspace\develop\file.h(340,78): warning C4239: nonstandard extension used: 'default argument': conversion from 'CSize' to 'CSize &' (compiling source file main.cpp) [C:\jenkins\workspace\develop\project.vcxproj]

```



```
175>c:\jenkins\workspace\develop\file.h(76): warning C4100: "Id": Unreferenzierter formaler Parameter (Quelldatei wird kompiliert main.cpp) [C:\jenkins\workspace\develop\project.vcxproj]

```

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/drulli.png" title="drulli's avatar" /> drulli:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-56450?focusedId=436011&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-436011">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Please do not reuse existing issues. Create a new one and attach the log in your console that is wrongly parsed to the result in your comment. It looks like the parser does not select the filename correctly.</p>
</pre>
</details>

</ul>
</details>

<p>Please do not reuse existing issues. Create a new one and attach the log in your console that is wrongly parsed to the result in your comment. It looks like the parser does not select the filename correctly.</p>
